electron-app/src/server/websites/bluesky/bluesky.service.ts

```diff
@@ -341,11 +348,17 @@ export class Bluesky extends Website {
     const isAutoscaling: boolean = submissionPart.data.autoScale;
 
     if (!submissionPart.data.altText) {
-      problems.push(`Bluesky requires alt text to be provided`);
+      warnings.push(`Bluesky recommends alt text to be provided`);
     }
 
-    if (submissionPart.data.description.value.length > this.MAX_CHARS) {
-      problems.push(`Max description length allowed is ${this.MAX_CHARS} characters.`);
+    const rt = new RichText({ text: submissionPart.data.description.value });
+    const agent = new BskyAgent({ service: 'https://bsky.social' })
+    rt.detectFacets(agent);
+
+    if (rt.graphemeLength > this.MAX_CHARS) {
+      problems.push(
+        `Max description length allowed is ${this.MAX_CHARS} graphemes.`,
```

Comment:
question: what exactly is a grapheme?

Reply:
A crazy unit that BlueSky has decided to use IMO, but basically its a phonetic character descriptive length metric. It can be one or more character.

Reply:
the most confusing thing that grapheme source code has 10k lines file

Reply:
https://github.com/flmnt/graphemer/blob/master/src/Graphemer.ts

Reply:
Yup. It's literal insanity IMO that they picked that and not a standard length metric. But ... hey, it's BlueSky. 🤷‍♂️

Reply:
weird af